ospf6d/ospf6_message.c

```diff
@@ -2356,7 +2356,22 @@ static void ospf6_send_lsupdate(struct ospf6_neighbor *on,
 	}
 	if (oi) {
 		ospf6_packet_add(oi, op);
-		OSPF6_MESSAGE_WRITE_ON(oi);
+		/* If ospf instance is being deleted, send the packet
+		 * immediately
+		 */
+		if ((oi->area == NULL) || (oi->area->ospf6 == NULL))
+			return;
+		if (oi->area->ospf6->inst_shutdown) {
+			struct thread os_packet_thd;
+
+			os_packet_thd.arg = (void *)oi->area->ospf6;
+			if (oi->on_write_q == 0) {
+				listnode_add(oi->area->ospf6->oi_write_q, oi);
+				oi->on_write_q = 1;
+			}
+			ospf6_write(&os_packet_thd);
```

Comment:
I think this could be smaller code if you use `thread_execute` instead of generating a `struct thread`. As a bonus you get slow function execution warnings.

Example:

```c
thread_execute(master, ospf6_write, oi->area->ospf6, 0);
```

Reply:
changed to thread_execute